microservices/embeddingservice/src/embed_text.py

```diff
@@ -43,37 +44,46 @@ def download_model(
 class EmbedText:
     def __init__(self, config):
         self.config = config
+        self.model_config = config["models"]["c2c_models"]
         self.models = {}
         bucket = None
         if sa := self.config.get("service_account"):
             credentials = service_account.Credentials.from_service_account_info(sa)
             client = storage.Client(credentials=credentials)
             bucket = client.bucket(self.config["bucket_name"])
-        for model in self.config["models"]:
-            for model_name, model_path in model.items():
-                bucket_path = (
-                    f'{self.config["bucket_path"]}/{model_path.split("/")[-1]}.zip'
-                )
+        for model in self.model_config:
+            if re.match(r"^opensearch://", self.model_config[model]["endpoint"]):
```

Comment:
I like to avoid using regex whenever possible, especially if it can be replaced by a standardlib function, in this case `self.model_config[model]["endpoint"].startswith("opensearch://")`

Apart from that, can you please explain why this check is neccessay in the first place?

Reply:
This is because there are 2 types of models: the ones in opensearch and the ones in the PA service. Both are needed in the reco explorer but only the opensearch models are needed to ingest the embeddings into the opensearch index.